hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsBackoffMetrics.java

```diff
@@ -0,0 +1,275 @@
+    setMetricValue(metric, value, null);
+  }
+
+  /*
+  Acronyms :-
+  1.RCTSI :- Request count that succeeded in x retries
+  2.MMA :- Min Max Average (This refers to the backoff or sleep time between 2 requests)
+  3.s :- seconds
+   */
+  private void getRetryMetrics(StringBuilder metricBuilder) {
+    for (RetryValue retryCount : RETRY_LIST) {
+      long totalRequests = getMetricValue(TOTAL_REQUESTS, retryCount);
+      metricBuilder.append("$RCTSI$_").append(retryCount.getValue())
+              .append("R=").append(getMetricValue(NUMBER_OF_REQUESTS_SUCCEEDED, retryCount));
+
+      if (totalRequests > 0) {
+        metricBuilder.append("$MMA$_").append(retryCount.getValue())
+                .append("R=").append(format(DOUBLE_PRECISION_FORMAT, (double) getMetricValue(MIN_BACK_OFF, retryCount) / THOUSAND)).append("s")
+                .append(format(DOUBLE_PRECISION_FORMAT, (double) getMetricValue(MAX_BACK_OFF, retryCount) / THOUSAND)).append("s")
+                .append(format(DOUBLE_PRECISION_FORMAT, ((double) getMetricValue(TOTAL_BACK_OFF, retryCount) / totalRequests) / THOUSAND)).append("s");
+      } else {
+        metricBuilder.append("$MMA$_").append(retryCount.getValue()).append("R=0s");
+      }
+    }
+  }
+
+  /*
+  Acronyms :-
+  1.BWT :- Number of Bandwidth throttled requests
+  2.IT :- Number of IOPS throttled requests
+  3.OT :- Number of Other throttled requests
+  4.NFR :- Number of requests which failed due to network errors
+  5.%RT :- Percentage of requests that are throttled
+  6.TRNR :- Total number of requests which succeeded without retrying
+  7.TRF :- Total number of requests which failed
+  8.TR :- Total number of requests which were made
+  9.MRC :- Max retry count across all requests
+   */
+  private void getBaseMetrics(StringBuilder metricBuilder) {
+    long totalRequestsThrottled = getMetricValue(NUMBER_OF_NETWORK_FAILED_REQUESTS)
+            + getMetricValue(NUMBER_OF_IOPS_THROTTLED_REQUESTS)
+            + getMetricValue(NUMBER_OF_OTHER_THROTTLED_REQUESTS)
+            + getMetricValue(NUMBER_OF_BANDWIDTH_THROTTLED_REQUESTS);
+
+    metricBuilder.append("$BWT=").append(getMetricValue(NUMBER_OF_BANDWIDTH_THROTTLED_REQUESTS))
+            .append("$IT=").append(getMetricValue(NUMBER_OF_IOPS_THROTTLED_REQUESTS))
+            .append("$OT=").append(getMetricValue(NUMBER_OF_OTHER_THROTTLED_REQUESTS))
+            .append("$RT=").append(formatPercent(totalRequestsThrottled/ (double) getMetricValue(TOTAL_NUMBER_OF_REQUESTS), 3))
+            .append("$NFR=").append(getMetricValue(NUMBER_OF_NETWORK_FAILED_REQUESTS))
+            .append("$TRNR=").append(getMetricValue(NUMBER_OF_REQUESTS_SUCCEEDED_WITHOUT_RETRYING))
+            .append("$TRF=").append(getMetricValue(NUMBER_OF_REQUESTS_FAILED))
+            .append("$TR=").append(getMetricValue(TOTAL_NUMBER_OF_REQUESTS))
+            .append("$MRC=").append(getMetricValue(MAX_RETRY_COUNT));
+  }
```

Comment:
here as well move all acronyms to constants file

Reply:
Sure, will move all acronyms to constant file